From a1afabd3ce740872a857234b91eeb64fcf239f6a Mon Sep 17 00:00:00 2001
From: Klaus Ethgen <Klaus@Ethgen.de>
Date: Sat, 27 Sep 2014 12:07:10 +0100
Subject: [PATCH] Fix lcms2 integration

Thanks Michael Schwendt <mschwendt@gmail.com> for the bug reporting and
the patch.
---
 src/color-man.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/color-man.c b/src/color-man.c
index 7788c7a..e4f1bab 100644
--- a/src/color-man.c
+++ b/src/color-man.c
@@ -432,10 +432,12 @@ static gchar *color_man_get_profile_name(ColorManProfileType type, cmsHPROFILE p
 			if (profile)
 				{
 #ifdef HAVE_LCMS2
-				cmsUInt8Number profileID[17];
-				profileID[16] = '\0';
-				cmsGetHeaderProfileID(profile, profileID);
-				return g_strdup((gchar *) profileID);
+				cmsUInt32Number r;
+				char buffer[20];
+				buffer[0] = '\0';
+				r = cmsGetProfileInfoASCII(profile, cmsInfoDescription, "en", "US", &buffer, 20);
+				buffer[19] = '\0'; /* Just to be sure */
+				return g_strdup(buffer);
 #else
 				return g_strdup(cmsTakeProductName(profile));
 #endif
-- 
1.7.1

